app/src/main/java/com/eveningoutpost/dexdrip/Models/Tomato.java

```diff
@@ -139,15 +158,15 @@ static void AreWeDone() {
         s_recviedEnoughData = true;
         
         boolean checksum_ok = JoH.LibreCrc(data);
-        Log.e(TAG, "We have all the data that we need checksum_ok = " + checksum_ok + HexDump.dumpHexString(data));
+        Log.e(TAG, "We have all the data that we need " + s_acumulatedSize + " checksum_ok = " + checksum_ok + HexDump.dumpHexString(data));
         
         if(!checksum_ok) {
-            return;
+            throw new RuntimeException(CHECKSUM_FAILED);
```

Comment:
IMHO, using exception makes the code harder to understand (especially when this are not real exception, but something that might happen and can create a retry).
I guess that in this case, usage of it is correct, but most xDrip code is not using exceptions but return values.
Up to you if to change it or not.

Reply:
I used an exception here so that we can catch the result from `AreWeDone()` in the grandparent `decodeTomatoPacket()` rather than using return codes or something like that. I considered creating an explicit Exception for the purpose but thought it best not to over-complicate things at this stage where we're only dealing with a single error type.